packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -303,13 +303,30 @@ class Scrollable extends StatefulWidget {
   /// Calling this method will create a dependency on the closest [Scrollable]
   /// in the [context], if there is one.
   ///
+  /// The optional [Axis] allows targeting of a specific [Scrollable] of that
+  /// axis, useful when Scrollables are nested. When [axis] is provided, the
+  /// nearest enclosing [ScrollableState] in that [Axis] is returned, or null
+  /// if there is none.
+  ///
   /// See also:
   ///
   /// * [Scrollable.of], which is similar to this method, but asserts
   ///   if no [Scrollable] ancestor is found.
-  static ScrollableState? maybeOf(BuildContext context) {
-    final _ScrollableScope? widget = context.dependOnInheritedWidgetOfExactType<_ScrollableScope>();
-    return widget?.scrollable;
+  static ScrollableState? maybeOf(BuildContext context, { Axis? axis }) {
+    if (axis == null) {
+      final _ScrollableScope? widget = context.dependOnInheritedWidgetOfExactType<_ScrollableScope>();
+      return widget?.scrollable;
+    }
+    ScrollableState? scrollable = context.getInheritedWidgetOfExactType<_ScrollableScope>()?.scrollable;
+    while (scrollable != null) {
+      if (axisDirectionToAxis(scrollable.axisDirection) == axis) {
+        scrollable = context.dependOnInheritedWidgetOfExactType<_ScrollableScope>()?.scrollable;
```

Comment:
This creates a dependency for the context of the previous scrollable (set in line 326) and not for the context passed into the method, as one would probably expect. So, the context passed into the method may not be informed to rebuild when the ScrollableState returned by this method is changing...

You probably want to use `getElementForInheritedWidgetOfExactType` to obtain the element you want to depend on and than use `dependOnInheritedElement` on the context passed into the method to establish the dependency between the right context and the inherited element.

Reply:
Would be great to add a test case for this scenario to make sure the dependency ends up on the right context!

Reply:
Oh! Great point, I will fix that.